kafka/src/main/java/com/metamx/tranquility/kafka/util/PropertiesHelper.java

```diff
@@ -0,0 +1,149 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.metamx.tranquility.kafka.util;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Sets;
+import com.metamx.common.IAE;
+import com.metamx.common.logger.Logger;
+import com.metamx.tranquility.kafka.model.TopicSchemaPair;
+import org.apache.commons.lang.Validate;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Read and validate application settings from a properties file.
+ */
+public class PropertiesHelper
+{
+  public static final String ZK_CONNECT = "zookeeper.connect";
+  public static final String ZK_SESSION_TIMEOUT_MS = "zookeeper.session.timeout.ms";
+
+  public static final String KAFKA_GROUP_ID = "kafka.group.id";
+  public static final String KAFKA_ZK_BASE_PATH = "kafka.zk.base.path";
+  public static final String KAFKA_NUM_THREADS = "kafka.consumer.numThreads";
+
+  public static final String DRUID_INDEXING_SERVICE_NAME = "druid.indexing.serviceName";
+  public static final String DRUID_CURATOR_PATH = "druid.discovery.curator.path";
+  public static final String TRANQUILITY_COMMIT_MS = "tranquility.commit.period.ms";
+
+  public static final String TRANQUILITY_SCHEMA_MAP = "tranquility.schema.map";
+
+  private static final Logger log = new Logger(PropertiesHelper.class);
+  private static final ObjectMapper mapper = new ObjectMapper();
+
+  public static Properties getApplicationArguments(String propertiesFile) {
+    Properties props = new Properties(getDefaults());
+
+    try (InputStream is = new FileInputStream(propertiesFile)) {
+      props.load(is);
+    } catch (IOException e) {
+      log.error("Could not read property file: %s, aborting.", propertiesFile);
+      Throwables.propagate(e);
+    }
+
+    validateProperties(props);
+
+    for (Map.Entry<?,?> entry : props.entrySet()) {
+      String key = (String) entry.getKey();
+      if (key.startsWith("druid.extensions.")) {
+        System.setProperty(key, (String) entry.getValue());
+      }
+    }
+
+    IngestionSpecHelper.configPath = propertiesFile;
```

Comment:
The fact that this is being set means that `IngestionSpecHelper` really wants to be a regular class (with a constructor that takes configPath) rather than a bunch of statics.